### PR TITLE
Clarify trailing index behaviour and commands interactions

### LINKED ISSUE #329: Clarify trailing index behaviour and commands interactions
Add in the fact that displayed index would takes precedent when talking about offscreen index in UG (Note on command formats): #287, #265, #267

Update unassign examples to make it less "attackable" (e.g mention if there's no assignables): #267

### PR DESCRIPTION
**Pull request by C5hives**
Resolves #329
 
### Changes
- Added nested `view` command bug into "Known Issues"
- Added explanation for changing list indexes under "Command Formats"
- Linked nested `view` command bug in section explaining off-screen operations.
- Linked the relevant sections for each command (e.g. `view` bug in `edit` command, changing indexes in `view` command)
- Added some safeguards for given examples.
- Also added image to explain what our indexing can look like for assignable and assigned lists
- General rephrasing and reformatting.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -364,6 +383,12 @@ Therefore, edit the data file only if you are confident that you can update it c
 
 1. **When using multiple screens**, if you move the application to a secondary screen, and later switch to using only the primary screen, the GUI will open off-screen. The remedy is to delete the `preferences.json` file created by the application before running the application again.
 2. **If you minimize the Help Window** and then run the `help` command (or use the `Help` menu, or the keyboard shortcut `F1`) again, the original Help Window will remain minimized, and no new Help Window will appear. The remedy is to manually restore the minimized Help Window.
+3. **Alternating between viewing vendors and events consecutively** can lead to unexpected behaviour, especially when used after a `find` command.
```
Comment (Jasperabez) [inline docs/UserGuide.md:386]: not sure whether there's a better wording than unexpected

**Review by Jasperabez — APPROVED** [review]
LGTM but made a small comment

File: docs/UserGuide.md (hunk 1 above)
Comment (C5hives) [reply docs/UserGuide.md:386]: "Issuing alternating view-vendor and view-event commands"?

Comment (C5hives) [reply docs/UserGuide.md:386]: Changed to "Switching back and forth between vendor and event views can lead to unexpected behaviour, especially when used after a `find` command."

**Pull request merged**